src/modules/roc_core/target_libunwind/roc_core/backtrace.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#define UNW_LOCAL_ONLY
+#include <libunwind.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "roc_core/backtrace.h"
+
+namespace roc {
+namespace core {
+
+namespace {
+
+enum { MaxDigits = 10, MaxLen = 200, MaxLenFunctionName = 128 };
+/* safe concatenation of strings
+ */
+void safe_strcat(char* buffer, size_t& buffer_size, const char* str) {
+    size_t len = strlen(str);
+	/* Checking if there is enough space in the buffer
+	 */
+    if (buffer_size + len < MaxLen) {
+        strcat(buffer, str);
+        buffer_size += len;
+    } else {
+        size_t length_to_be_truncated = buffer_size + len - MaxLen;
+        for (size_t i = 0; i < len - length_to_be_truncated && buffer_size < MaxLen-1; i++) {
+            buffer[buffer_size] = str[i];
+            buffer_size++;
+        }
+        buffer[MaxLen - 1] = '\n';
```

Comment:
I guess we need `\0` instead of `\n`.

But I suggest to simplify the function and remove this piece at all:

```
void safe_strcat(char* buffer, size_t& buffer_size, const char* str) {
    size_t len = strlen(str);
    if (len > MaxLen - buffer_size - 1) {
        len = MaxLen - buffer_size - 1;
    }
    if (len > 0) {
        strncat(buffer, str, len);
        buffer_size += len;
    }
}
```